### PR TITLE
Auth functionality

### PR DESCRIPTION
**Pull request by sameep-s**
**Whats Added**

- Private Routes.
- Login Signup Functionlaity.

### CODE REVIEW

File: src/components/Footer/footer.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/* -----------------------Footer CSS */
+
+.container__main__footer{
+    min-height: 40vh;
+    width: 100%;
+    background-color: #f5f5f7;
```
Comment (parulgupta2001) [inline src/components/Footer/footer.css:6]: You should use var here

File: src/components/FooterLinkContainer/linkContainer.css [hunk 2]
```diff
@@ -0,0 +1,20 @@
+
+.footer__links__heading{
+    font-size: 2rem;
+    font-family: var(--font-heading);
+    font-weight: bolder;
+    color: #1d1d1f;
```
Comment (parulgupta2001) [inline src/components/FooterLinkContainer/linkContainer.css:6]: here also use var

**Comment by sanketr43** [comment]
.env file is supposed to be a private file, which is only hosted on a live server. You should not commit it in the git. It contains private data related to projects which get exposed to the public. Other things are fine just remove ENV and add it to gitignore file.

File: src/pages/CardProduct.jsx [hunk 3]
```diff
@@ -1,9 +1,11 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import '../pages-css/props.css';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faHeart } from "@fortawesome/free-solid-svg-icons";
 import { useCart } from '../util_Contexts';
 import { useWishlist } from '../util_Contexts/wishlist-context';
+import { useAuth } from '../util_Contexts/auth-context';
```
Comment (parulgupta2001) [inline src/pages/CardProduct.jsx:7]: You should export all contexts at one place eg- index.js and then export them.

**Review by parulgupta2001 — APPROVED** [review]
some suggestions are suggested otherwise everything looks good.

**Comment by sameep-s** [comment]
> .env file is supposed to be a private file, which is only hosted on a live server. You should not commit it in the git. It contains private data related to projects which get exposed to the public. Other things are fine just remove ENV and add it to gitignore file.

My functionality wasn't working without it so this is a temporary solution as submission is tomorrow, i'll fix this later.

File: src/pages/CardProduct.jsx (hunk 3 above)
Comment (sameep-s) [reply src/pages/CardProduct.jsx:7]: sure i have done like that in my other project, this one i'll have to change imports in every file so i'll fix this after submission. thanks for pointing out anyways.

File: src/components/Footer/footer.css (hunk 1 above)
Comment (sameep-s) [reply src/components/Footer/footer.css:6]: okay i'll fix that.

**Comment by sanketr43** [comment]
> > .env file is supposed to be a private file, which is only hosted on a live server. You should not commit it in the git. It contains private data related to projects which get exposed to the public. Other things are fine just remove ENV and add it to gitignore file.
> 
> My functionality wasn't working without it so this is a temporary solution as submission is tomorrow, i'll fix this later.

If you are using varcel or netlify then you can set the environment variables in the project settings directly. Your hosted project will access those variable directly.

**Review by sanketr43 — APPROVED** [review]

**Pull request merged**